### ISSUE TITLE
spinners aren't contained in progres bar on bottom of the screen

### ISSUE THREAD

**Issue by ticotaco72** [ppy/osu#2992]
in places where spinners should be seen on progres bar there's a blank space
good visible in start of this beatmap: https://osu.ppy.sh/beatmapsets/323961#osu/719902
logs:
[database.log](https://github.com/ppy/osu/files/2181325/database.log)
[network.log](https://github.com/ppy/osu/files/2181326/network.log)
[performance.log](https://github.com/ppy/osu/files/2181327/performance.log)
[performance-gamethread.audio.log](https://github.com/ppy/osu/files/2181328/performance-gamethread.audio.log)
[performance-gamethread.draw.log](https://github.com/ppy/osu/files/2181329/performance-gamethread.draw.log)
[performance-gamethread.input.log](https://github.com/ppy/osu/files/2181330/performance-gamethread.input.log)
[performance-gamethread.update.log](https://github.com/ppy/osu/files/2181331/performance-gamethread.update.log)
[runtime.log](https://github.com/ppy/osu/files/2181332/runtime.log)

**Comment by HoutarouOreki** [comment]
Spinners are contained in the progress bar though (see [Ieva's Polka Spinner Map](https://osu.ppy.sh/beatmapsets/57318#osu/172822)).

The reason why the progress is bottom flat, is because (I assume) object density is very low in these parts. See [FREEDOM DiVE - xi / R3](https://osu.ppy.sh/beatmapsets/761041#osu/1600415) at 1:20. There are circles, and yet it's flat.

**Comment by peppy** [comment]
The display needs a better scaling/display algorithm. Right now it is basically placeholder-level.

**Comment by ticotaco72** [comment]
do you think something like that is unexcepted too?
![osu_2018-07-09_23-46-46](https://user-images.githubusercontent.com/32620809/42738173-ab20bbea-887f-11e8-8888-d7cb00b9e438.jpg)
![osu_2018-07-15_22-24-51](https://user-images.githubusercontent.com/32620809/42738174-ab4429d6-887f-11e8-86ef-3bdfce099030.jpg)

**Comment by HoutarouOreki** [comment]
As been said, for now the progress bar is just a placeholder, it will be redone properly and thoroughly in the future. It's kinda meant not to work 100% for now.

**Comment by peppy** [comment]
Yes, that is another example of incorrect scaling. For instance, I would expect at least one block to be highlighted when there are any objects at all (versus none).

We welcome contributions if you'd like to improve on it!

**Comment by ticotaco72** [comment]
last comment was updated
and how can i reach informations about overall beatmap difficulty and meta data about it?
i wanted to ask how diificulty is calculated from which values or where in the source code, then i can read myself.

**Comment by HoutarouOreki** [comment]
Wouldn't changing this https://github.com/ppy/osu/blob/d0f7bdbcb8c8b6558185630341e0e047360fe4ff/osu.Game/Screens/Play/SquareGraph.cs#L170
 to `private float cubeCount => Max(DrawHeight / WIDTH, 1)` work fine for now?

**Comment by ticotaco72** [comment]
there's 'no max' exception while building, so not

**Comment by HoutarouOreki** [comment]
Math.Max()

**Comment by ticotaco72** [comment]
it doesn't help. as ppy said i think there's a need to rewrite it

and i have last question about how to reach info about sliders and spinners properties? are they included in `osu.Game.Rulesets.Objects.HitObject.cs`

and sorry for all this questions. it's my first contribution so i wanna know it to make good code, and for not knowing some things like above that max is in math library(this easy to predict), because normally i worked in python where everything is in main namespace.
